src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -15,9 +15,83 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     /// A simple class that combines a single type symbol with annotations
     /// </summary>
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-    internal abstract class TypeSymbolWithAnnotations : IFormattable
+    internal readonly struct TypeSymbolWithAnnotations : IFormattable
     {
-        public sealed override string ToString() => TypeSymbol.ToString();
+        /// <summary>
+        /// A builder for lazy instances of TypeSymbolWithAnnotations.
+        /// </summary>
+        [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
+        internal struct Builder
+        {
+            private TypeSymbol _defaultType;
+            private INonNullTypesContext _nonNullTypesContext;
+            private bool _isAnnotated;
+            private Extensions _extensions;
+
+            internal TypeSymbol DefaultType => _defaultType;
```

Comment:
>DefaultType [](start = 32, length = 11)

Consider xml doc on the builder's properties as well. That'll be useful for QuickInfo in code.